plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2531,6 +2554,29 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
         commitTransaction(transaction, "set table properties");
     }
 
+    public static Map<String, String> calculateTableCompressionProperties(IcebergFileFormat oldFileFormat, IcebergFileFormat newFileFormat, Map<String, String> existingProperties, Map<String, Object> inputProperties)
+    {
+        ImmutableMap.Builder<String, String> newCompressionProperties = ImmutableMap.builder();
+
+        Optional<HiveCompressionCodec> oldCompressionCodec = getHiveCompressionCodec(oldFileFormat, existingProperties);
+        Optional<HiveCompressionCodec> newCompressionCodec = IcebergTableProperties.getHiveCompressionCodec(inputProperties);
+
+        Optional<HiveCompressionCodec> compressionCodec = newCompressionCodec.or(() -> oldCompressionCodec);
```

Comment:
Nit: no need for a lambda here, just `newCompressionCodec.orElse(oldCompressionCodec)`

Reply:
orElse requires the parameter to be T and return T, but we are processing on optional<T> and want to return optional<T>

Reply:
Ah, I missed that- still, you want to avoid spinning a lambda for a cheap operation like this. Maybe a cleaner way to write it is: `newCompressionCodec.isPresent() ? newCompressionCodec : oldCompressionCodec;` like we do later on for compression level.